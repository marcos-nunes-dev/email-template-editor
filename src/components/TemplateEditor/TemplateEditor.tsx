import React, { ReactElement } from 'react';
import { CodeEditor } from 'components/CodeEditor';
import { useTemplate } from 'hooks/useTemplate';
import { useRouter } from 'next/router';

export function TemplateEditor(): ReactElement {
    const { code, setCode } = useTemplate();
    const { query } = useRouter();
    return (
        <CodeEditor
            initialCode={code}
            codeEditHandler={setCode}
            fileName={query.template_name}
        />
    );
}
