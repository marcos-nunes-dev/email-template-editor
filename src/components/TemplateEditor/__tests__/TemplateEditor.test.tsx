/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from 'tests';
import { TemplateEditor } from '../TemplateEditor';

describe('<TemplateEditor />', () => {
    it('should match snapshot', () => {
        const { container } = render(<TemplateEditor />);
        expect(container).toMatchSnapshot();
    });
});
