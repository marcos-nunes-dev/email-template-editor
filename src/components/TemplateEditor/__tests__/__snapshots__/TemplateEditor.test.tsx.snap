// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateEditor /> should match snapshot 1`] = `
<div>
  <div
    class="Box-sc-badax0-0 cplrVY"
    height="inherit"
  >
    <div
      class="Box-sc-badax0-0 iZEXDH"
    >
      <div
        class="Box-sc-badax0-0 kdTlVT"
        display="flex"
        width="fit-content"
      >
        <div
          class="Box-sc-badax0-0 iLKvW"
          height="5px"
          width="5px"
        />
        <div
          class="Box-sc-badax0-0 Text__TextStyled-sc-1u8mq76-0 eExxtX jJTLlT"
          color="black"
          font-size="sm"
          font-weight="bold"
        >
          undefined.tmp
        </div>
      </div>
    </div>
    <div
      class="Box-sc-badax0-0 styles__EditorWrapper-sc-1rehzf6-0  iTPVtL"
    >
      <div
        class="code-editor"
        style="position: relative; text-align: left; box-sizing: border-box; padding: 0px; overflow: hidden;"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="npm__react-simple-code-editor__textarea"
          data-gramm="false"
          id="code-editor"
          spellcheck="false"
          style="margin: 0px; border: 0px; background: none; box-sizing: inherit; display: inherit; font-family: inherit; font-style: inherit; font-variant-ligatures: inherit; font-weight: inherit; letter-spacing: inherit; line-height: inherit; tab-size: inherit; text-indent: inherit; text-rendering: inherit; text-transform: inherit; white-space: pre-wrap; word-break: keep-all; overflow-wrap: break-word; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; resize: none; overflow: hidden; padding: 10px 10px 10px 10px;"
        />
        <pre
          aria-hidden="true"
          style="margin: 0px; border: 0px; background: none; box-sizing: inherit; display: inherit; font-family: inherit; font-style: inherit; font-variant-ligatures: inherit; font-weight: inherit; letter-spacing: inherit; line-height: inherit; tab-size: inherit; text-indent: inherit; text-rendering: inherit; text-transform: inherit; white-space: pre-wrap; word-break: keep-all; overflow-wrap: break-word; position: relative; pointer-events: none; padding: 10px 10px 10px 10px;"
        >
          <br />
        </pre>
        <style
          type="text/css"
        >
          
/**
 * Reset the text fill color so that placeholder is visible
 */
.npm__react-simple-code-editor__textarea:empty {
  -webkit-text-fill-color: inherit !important;
}

/**
 * Hack to apply on some CSS on IE10 and IE11
 */
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  /**
    * IE doesn't support '-webkit-text-fill-color'
    * So we use 'color: transparent' to make the text transparent on IE
    * Unlike other browsers, it doesn't affect caret color in IE
    */
  .npm__react-simple-code-editor__textarea {
    color: transparent !important;
  }

  .npm__react-simple-code-editor__textarea::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}

        </style>
      </div>
    </div>
  </div>
</div>
`;
